src/Sulu/Component/SmartContent/Orm/DataProviderRepositoryTrait.php

```diff
@@ -56,23 +56,23 @@ public function findByFilters($filters, $page, $pageSize, $limit, $locale, $opti
      */
     private function findByFiltersIds($filters, $page, $pageSize, $limit, $locale, $options = [])
     {
-        $parameter = [];
+        $queryBuilder = $this->createQueryBuilder('c');
 
-        $queryBuilder = $this->createQueryBuilder('c')
-            ->select('c.id')
-            ->distinct()
-            ->orderBy('c.id', 'ASC');
+        $parameter = $this->appendSelect($queryBuilder, 'c', $locale, $options);
 
         $tagRelation = $this->appendTagsRelation($queryBuilder, 'c');
         $categoryRelation = $this->appendCategoriesRelation($queryBuilder, 'c');
 
-        if (array_key_exists('sortBy', $filters) && is_array($filters['sortBy'])) {
+        if (array_key_exists('sortBy', $filters)) {
             $sortMethod = array_key_exists('sortMethod', $filters) ? $filters['sortMethod'] : 'asc';
-            $this->appendSortBy($filters['sortBy'], $sortMethod, $queryBuilder, 'c', $locale);
+            $sortColumn = $filters['sortBy'];
 
-            foreach ($filters['sortBy'] as $sortColumn) {
-                $queryBuilder->addSelect($sortColumn);
+            if ($sortColumn === explode('.', $sortColumn)[0]) {
```

Comment:
What are you checking for here?

Reply:
checking if `$sortColumn` contains a `.` ... just realized i could use `strpos` for that.

Reply:
Actually I was asking why you were checking if it contains a `.` :slightly_smiling_face:

Reply:
So I can prepend the `sortColumn` with the `alias`.